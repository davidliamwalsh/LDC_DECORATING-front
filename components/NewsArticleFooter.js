import { Component } from 'react'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'
import moment from 'moment'

class NewsArticleFooter extends Component {

  constructor (props) {
    super(props)

    this.NEWSARTICLES_QUERY = gql`
      query newsArticles {
        newsArticles  {
          slug
          id
          title
          body
          createdAt
        }
      }
    `
  }

  render () {
    const { newsArticleSize } = this.props

    return <Query query={this.NEWSARTICLES_QUERY}>
      {({ loading, data }) => {
        if (loading) {
          return <>
            <p>Loading...</p>
          </>
        } else {
          return <>
            {data.newsArticles.slice(0, newsArticleSize).map((newsArticle, index) => {
              return <li className="c-footer--upper__list--item" key={index}>
                <div className="c-footer--upper__list--item__news-article">
                  <div className="c-footer--upper__list--item__news-article--date">
                    <h5 className="c-footer--upper__list--item__news-article--date__item">{moment(newsArticle.createdAt).format('MMM Do')}</h5>
                  </div>
                  <div className="c-footer--upper__list--item__news-article--main">
                    <h5 className="c-footer--upper__list--item__news-article--main__title">{newsArticle.title}</h5>
                    <h5 className="c-footer--upper__list--item__news-article--main__body c-footer--upper__list--item-lineheight">{newsArticle.body}</h5>
                  </div>
                </div>
              </li>
            })}
          </>
        }
      }}
    </Query>
  }
}

export default NewsArticleFooter