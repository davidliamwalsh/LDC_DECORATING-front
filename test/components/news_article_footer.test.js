import NewsArticleFooter from '../../components/NewsArticleFooter'

describe('<NewsArticleFooter />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <NewsArticleFooter />
    )

    expect(component).toMatchSnapshot()
  })
})