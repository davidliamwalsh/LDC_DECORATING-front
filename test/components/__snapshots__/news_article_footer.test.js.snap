// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsArticleFooter /> it renders correctly 1`] = `ShallowWrapper {}`;
